src/route-loading.ts

```diff
@@ -0,0 +1,145 @@
+import { activationStrategy, _buildNavigationPlan } from './navigation-plan';
+import { NavigationInstruction } from './navigation-instruction';
+import { ViewPortInstruction, RouteConfig, RoutingComponent } from './interfaces';
+import { Redirect } from './navigation-commands';
+import { Router } from './router';
+import { Next } from './pipeline';
+
+export class RouteLoader {
+  loadRoute(router: Router, config: RouteConfig, navigationInstruction: NavigationInstruction): Promise<RoutingComponent> {
+    throw Error('Route loaders must implement "loadRoute(router, config, navigationInstruction)".');
+  }
+}
+
+export class LoadRouteStep {
+
+  static inject() { return [RouteLoader]; }
+  /**@internal */
+  routeLoader: RouteLoader;
+
+  constructor(routeLoader: RouteLoader) {
```

Comment:
is there an advantage to @internal over 

`constructor(private routeLoader: RouteLoader)`?

Reply:
if we used `private`, the `d.ts` will still have it, while `@internal` strips it entirely from the generated result. It doesn't really matter much unless we have a lot of comments for `private` for better understanding from interfaces